File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let newStr = ""
  for (let i = str.length - 1; i >= 0; i--) {
    newStr = newStr + str[i]
  }
  return newStr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("Expecting: 'yob gib'");
  console.log("=>", reverseString("big boy"))
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

/**
 * let newString;
 * 
 * for (last index of string; up until index = 0; remove one from index) {
 *  add letter to newString
 * }
 * 
 * return newString
 *
 */

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  newString = ""
  str.chars.each {|char| newString = char + newString}
  newString
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!

  puts "Expecting: 'yob gib'"
  puts "=>", reverse_string('big boy')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

# newString = ""
# for every char in string
#   newString = char + newString behind
# end
# newString